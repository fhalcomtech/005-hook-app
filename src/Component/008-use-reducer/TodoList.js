import React from 'react';
import TodoListItem from "./TodoListItem";

const TodoList = ({todos, handlerClickDone, handlerClickDelete}) => {
    return (
        <div className="col-7">
            <ul className="list-group list-group-flush">
                {todos.map((todo, index) =>
                    (<TodoListItem
                        todo={todo}
                        index={index}
                        handlerClickDelete={handlerClickDelete}
                        handlerClickDone={handlerClickDone}
                        key={todo.id}
                    />)
                )}
            </ul>
        </div>

    );
};

export default TodoList;
