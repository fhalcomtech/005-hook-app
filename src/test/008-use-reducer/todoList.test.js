import React from "react";
import { shallow } from "enzyme";
import TodoList from "../../Component/008-use-reducer/TodoList";
import { initialTodoState } from "../fixtures";

describe("Testing TodoList",()=> {
    const handlerClickDone = jest.fn(); 
    const handlerClickDelete = jest.fn();
    const wrapper = shallow(<TodoList todos = {initialTodoState} handlerClickDone = {handlerClickDone} handlerClickDelete = {handlerClickDelete}/>)
    
    test("Testing TodoList Snapshot", ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test("Testing TodoList count lisItems", () => {
        const listItems = wrapper.find("li").html();
        console.log(listItems.length);
    });
});