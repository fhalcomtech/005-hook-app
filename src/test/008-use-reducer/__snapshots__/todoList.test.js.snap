// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoList Testing TodoList  1`] = `
<div
  className="col-7"
>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={0}
      todo={
        Object {
          "desc": "Learn React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={1}
      todo={
        Object {
          "desc": "Learn Root",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;

exports[`Testing TodoList Testing TodoList Snapshot 1`] = `
<div
  className="col-7"
>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={0}
      todo={
        Object {
          "desc": "Learn React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={1}
      todo={
        Object {
          "desc": "Learn Root",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;

exports[`Testing TodoList Testing TodoList Snapshot 2`] = `
<div
  className="col-7"
>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={0}
      todo={
        Object {
          "desc": "Learn React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handlerClickDelete={[MockFunction]}
      handlerClickDone={[MockFunction]}
      index={1}
      todo={
        Object {
          "desc": "Learn Root",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;
